ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineContext.java

```diff
@@ -111,7 +112,7 @@ EngineContext createSandbox(final ServiceContext serviceContext) {
     persistentQueries.forEach((queryId, query) ->
         sandBox.persistentQueries.put(
             query.getQueryId(),
-            query.copyWith(sandBox::unregisterQuery)));
+            SandboxedPersistentQueryMetadata.of(query, sandBox::unregisterQuery)));
```

Comment:
can we add test coverage here for what should/shouldn't happen? we have test coverage for the sandbox, but not that on validation we don't actually affect the KS app

Reply:
Done.

I added the cases in `KsqlEngineTest`. I didn't use the `execute` method in the tests because it wasn't allowing me to execute the terminate 'cause it is a not executable statement.